spyder/plugins/application/widgets/install.py

```diff
@@ -250,11 +250,15 @@ def _download_install(self):
 
             url = ('https://github.com/spyder-ide/spyder/releases/latest/'
                    f'download/{name}')
-            destination = os.path.join(tmpdir, name)
-            logger.debug(f"Downloading installer from: {url}")
-            download = urlretrieve(url,
-                                   destination,
-                                   reporthook=self._progress_reporter)
+            path_destination = os.path.join(tmpdir, 'spyder',
+                                            self.latest_release_version)
+            os.makedirs(path_destination, exist_ok=True)
+            destination = os.path.join(path_destination, name)
+            if (not os.path.isfile(destination)):
+                logger.debug(f"Downloading installer from: {url}")
+                download = urlretrieve(url,
+                                       destination,
+                                       reporthook=self._progress_reporter)
```

Comment:
Lets use some more self explanatory variables names here, add another intermidiate level dir (`updates`) and also add more info about where the executable gets downloaded to the debug log entry. So this will end up something like:

```suggestion
            installer_dir_path = os.path.join(tmpdir, 'spyder', 'updates',
                                              self.latest_release_version)
            os.makedirs(installer_dir_path, exist_ok=True)
            installer_path = os.path.join(installer_dir_path, name)
            if (not os.path.isfile(installer_path)):
                logger.debug(f"Downloading installer from {url} to {installer_path}")
                download = urlretrieve(url,
                                       installer_path,
                                       reporthook=self._progress_reporter)
```

Reply:
Also, I think we need to have a way to remove old executable versions downloaded. Probably, besides the validation to do or not the download, we also need to validate if there are more folders under the `spyder/updates` dir and remove them if they are not the latest release. What do you think @jsbautista ?

Reply:
> Also, something that is still pending from this whole new update flow is the division of the download and install process so users can decide if them want to proceed with the installation right now, when Spyder closes or not proceed at all, and how we should handle the case when the user says no, closes Spyder and launches it again.
> 
> Since that part has some initial ideas but probably will need more discussion and feedback I would say that we should leave it for a follow-up PR. Is that okay @mrclary @CAM-Gerlach @ccordoba12 ?

I think that is a good idea. This will require new/updated widgets as well as refactored logic. That will give us more time to discuss and finalize the UX that we want. A follow-up PR is my recommendation as well.

Reply:
@dalthviz I think that this could work very well, besides that with this there will be a maximum of one folder with the latest version checked

Reply:
> Since that part has some initial ideas but probably will need more discussion and feedback I would say that we should leave it for a follow-up PR. Is that okay @mrclary @CAM-Gerlach @ccordoba12 ?

Sure, though we really should get the UX in a good state before we actually release 5.4.0 to users, which as I understand is the plan here.

> Also, I think we need to have a way to remove old executable versions downloaded. Probably, besides the validation to do or not the download, we also need to validate if there are more folders under the spyder/updates dir and remove them if they are not the latest release. What do you think @jsbautista ?

Could be a good idea to at least cache the current/previous release in case something goes wrong and the user wants to roll back to the same version as before; that's what I see most other apps that do this do.